src/extension.ts

```diff
@@ -196,9 +200,49 @@ function registerDockerClients(): void {
         if (e.affectsConfiguration('docker.composeCommand')) {
             composeClient.reconfigure();
         }
+
+        if (e.affectsConfiguration('docker.contexts.showInStatusBar')) {
```

Comment:
Along with moving to another file, we can do another event registration. Overall it makes for very slightly less efficient code, but much neater.